tsconfig.json

```diff
@@ -15,14 +15,19 @@
     "moduleResolution": "node",
     "experimentalDecorators": true,
     "importHelpers": true,
-    "target": "es2015",
+    "target": "ES2021",
     "typeRoots": ["node_modules/@types"],
-    "lib": ["es2018", "dom"],
+    "lib": ["es2020", "dom"],
     "resolveJsonModule": true
   },
   "angularCompilerOptions": {
     "strictInjectionParameters": true,
     "strictInputAccessModifiers": true,
     "strictTemplates": true
-  }
+  },
+  "references": [
+    {
+      "path": "./src/tsconfig.app.json"
+    }
+  ]
```

Comment:
OOC, what is the benefit of using a project reference?

Reply:
It allows you to structure your TypeScript programs into smaller pieces. By doing this, you can greatly improve build times, enforce logical separation between components, and organize your code in new and better ways. [Source](https://www.typescriptlang.org/docs/handbook/project-references.html)

Reply:
The benefit of this can be that as a developer you can see all `tsconfig` configuration references for a given application from a single file. After all, you can have many `tsconfig` configurations